x/nodes/keeper/validator_test.go

```diff
@@ -150,6 +155,60 @@ func Test_sortNoLongerStakedValidators(t *testing.T) {
 	}
 }
 
+// Verifies the marshaled product of a validator is the same regardless of
+// the order of RewardDelegators.  If it is not, a key of the merkle tree in
+// application.db is not deterministic, that will cause consensus failure.
+func Test_Marshal_RewardDelegators(t *testing.T) {
+	numOfIterations := 100
+	numOfDelegators := 100
+
+	chains := []string{"0001", "0002", "005A", "005B", "005C", "005D"}
+	url := "https://test.pokt.network:443"
+	stake := int64(18000000000)
+	pubKey, _ := crypto.NewPublicKey("0b787e54e66b3db3a3396c2322d8314287e08990f7696c490153b078bada7e94")
+	nodeAddr := sdk.Address(pubKey.PubKey().Address())
+	outputAddr, _ := sdk.AddressFromHex("42846261e1798fc08e1dfd97325af7b280f815b0")
+
+	randNums := make([]int, numOfDelegators)
+	for j := 0; j < numOfDelegators; j++ {
+		randNums[j] = rand.Int()
+	}
+
+	var valHash string
+
+	for i := 0; i < numOfIterations; i++ {
+		rand.Shuffle(len(randNums), func(i, j int) {
```

Comment:
#PUC (#PleaseUpdateComment) on why we are shuffling.

Context: Note that the reason I ask fo these things isn't because I can't figure it out myself, but because I take time to think through it to understand. When the author adds a comment while implementing it, it saves time for many devs (and reviewers) many times over.

Reply:
Sure.  Added a comment about this.